e2e/runner/zevm.go

```diff
@@ -23,6 +25,64 @@ import (
 
 var gasLimit = big.NewInt(250000)
 
+// ApproveETHZRC20 approves ETH ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveETHZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.ETHZRC20)
+}
+
+// ApproveERC20ZRC20 approves ERC20 ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveERC20ZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.ERC20ZRC20)
+}
+
+// ApproveBTCZRC20 approves BTC ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveBTCZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.BTCZRC20)
+}
+
+// ApproveSOLZRC20 approves SOL ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveSOLZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.SOLZRC20)
+}
+
+// ApproveSPLZRC20 approves SPL ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveSPLZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.SPLZRC20)
+}
+
+// ApproveSUIZRC20 approves SUI ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveSUIZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.SUIZRC20)
+}
+
+// ApproveFungibleTokenZRC20 approves Sui fungible token ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveFungibleTokenZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.SuiTokenZRC20)
+}
+
+// ApproveTONZRC20 approves TON ZRC20 on EVM to a specific address
+func (r *E2ERunner) ApproveTONZRC20(allowed ethcommon.Address) {
+	r.approveZRC20(allowed, r.TONZRC20)
+}
```

Comment:
Wondering why cant we have one function which accepts the ZRC20 address as a second parameter

Reply:
Yeah we could eventually make this change, ApproveETHZRC20 and ApproveERC20ZRC20 are used extensively through the tests so I think it might be nice to have this function directly, I added the others for consistency but they're almost used once so I would be ok removing all of those